package/MDAnalysis/lib/formats/libmdaxdr.pyx

```diff
@@ -513,12 +475,18 @@ cdef class TRRFile(_XDRFile):
         cdef float time = 0
         cdef float lmbda = 0
 
-        # Use this instead of memviews here to make sure that references are
-        # counted correctly
-        cdef np.ndarray xyz = np.empty((self.n_atoms, DIMS), dtype=DTYPE)
-        cdef np.ndarray velocity = np.empty((self.n_atoms, DIMS), dtype=DTYPE)
-        cdef np.ndarray forces = np.empty((self.n_atoms, DIMS), dtype=DTYPE)
-        cdef np.ndarray box = np.empty((DIMS, DIMS), dtype=DTYPE)
+        cdef np.npy_intp[2] dim
+        dim[0] = self.n_atoms
+        dim[1] = DIMS
+
+        cdef np.npy_intp[2] unitcell_dim
+        unitcell_dim[0] = DIMS
+        unitcell_dim[1] = DIMS
+
+        cdef np.ndarray[np.float32_t, ndim=2] xyz = np.PyArray_EMPTY(2, dim, np.NPY_FLOAT32, 0)
```

Comment:
can likely allocate this once per Reader, unless natoms is changing?

Reply:
I tried and broke a bunch of tests and couldn't figure out why. I will come back to it.

Reply:
If we are deprecating anyway should we even bother?